src/stomach.cpp

```diff
@@ -170,6 +180,36 @@ units::volume stomach_contents::capacity( const Character &owner ) const
     if( owner.has_trait( trait_SLIMESPAWNER ) ) {
         max_mod *= 3;
     }
+
+    //Huge, for example, makes you roughly x2 larger, so stomach size is doubled
+    //The scientifically correct approach would be x8 stomach size, due to square-cube law, but that would break the game
+    //The same 'square-cube law is ignored for balance' reasoning is applied to the other size category mutations
+    // -ungen
+
+    //else if because they're mutually exclusive, that way we save a lot of uneccesary checks -ungen
+    if( owner.has_trait( trait_SMALL_OK ) ) {
+        max_mod *= 0.5;
+    }
+    else if( owner.has_trait( trait_SMALL2 ) ) {
+        max_mod *= 0.5;
+    }
+    else if( owner.has_trait( trait_SMALL ) ) {
+        max_mod *= 0.75;
+    }
+    else if( owner.has_trait( trait_LARGE ) ) {
+        max_mod *= 1.5;
+    }
+    else if( owner.has_trait( trait_LARGE_OK ) ) {
+        max_mod *= 1.5;
+    }
+    else if( owner.has_trait( trait_HUGE ) ) {
+        max_mod *= 2;
+    }
+    else if( owner.has_trait( trait_HUGE_OK ) ) {
+        max_mod *= 2;
+    }
+    //I thought this would be a lot harder to code, boy I was wrong -ungen
```

Comment:
I would be better if this is coded as a JSON attribute of traits instead of hardcoding it here.

Reply:
That's a good second PR idea, actually.